src/core/qgstemporalnavigationobject.cpp

```diff
@@ -0,0 +1,183 @@
+/***************************************************************************
+                         qgstemporalnavigationobject.cpp
+                         ---------------
+    begin                : March 2020
+    copyright            : (C) 2020 by Samweli Mwakisambwe
+    email                : samweli at kartoza dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgstemporalnavigationobject.h"
+#include "qgsproject.h"
+#include "qgsprojecttimesettings.h"
+#include "qgsmaplayer.h"
+#include "qgsrasterlayer.h"
+
+QgsTemporalNavigationObject::QgsTemporalNavigationObject()
+{
+}
+
+QDateTime QgsTemporalNavigationObject::addToDateTime( QDateTime datetime, QString time, int value )
```

Comment:
time should be a QgsInterval value here, then you can just use

    return datetime + time;